dvc/commands/data.py

```diff
@@ -0,0 +1,339 @@
+
+from dvc.cli.command import CmdBase
+from dvc.cli.utils import fix_subparsers
+from dvc.ui import ui
+
+if TYPE_CHECKING:
+    from dvc.output import Output
+    from dvc_data.hashfile.obj import HashFile
+
+
+logger = logging.getLogger(__name__)
+
+
+def _in_cache(obj, cache):
+    from dvc_objects.errors import ObjectFormatError
+
+    if not obj:
+        return False
+
+    try:
+        cache.check(obj.hash_info.value)
+        return True
+    except (FileNotFoundError, ObjectFormatError):
+        return False
+
+
+def shallow_diff(
+    root: str,
+    old_obj: Optional["HashFile"],
+    new_obj: Optional["HashFile"],
+    cache,
+):
+    # TODO: add support for shallow diff in dvc-data
+    # TODO: we may want to recursively do in_cache check
+    d = {}
+    if not _in_cache(old_obj, cache):
+        d["not_in_cache"] = [root]
+
+    from dvc_data.objects.tree import Tree
+
+    if isinstance(new_obj, Tree):
+        root = os.path.sep.join([root, ""])
+
+    if old_obj is None and new_obj is None:
+        return d
+    if old_obj is None:
+        return {"added": [root], **d}
+    if new_obj is None:
+        return {"deleted": [root], **d}
+    if old_obj.hash_info != new_obj.hash_info:
+        return {"modified": [root], **d}
+    return {"unchanged": [root], **d}
+
+
+def granular_diff(
+    root: str,
+    old_obj: Optional["HashFile"],
+    new_obj: Optional["HashFile"],
+    cache,
+):
+    from dvc_data.diff import ROOT
+    from dvc_data.diff import diff as odiff
+
+    def path_join(root, *paths):
+        if paths == ROOT:
+            return root
+        return os.path.sep.join([root, *paths])
+
+    diff = odiff(old_obj, new_obj, cache)
+    output = defaultdict(list)
+    for state in ("added", "deleted", "modified", "unchanged"):
+        items = getattr(diff, state)
+        output[state].extend(
+            path_join(root, *item.new.key)
+            for item in items
+            if item.new.key != ROOT
+        )
+        # TODO: PERF: diff is checking not_in_cache for each even if we only
+        # need it for the index.
+        # BUG: not_in_cache file also shows up as modified in staged and
+        # unstaged. We currently don't know if it is really modified.
+        output["not_in_cache"].extend(
+            path_join(root, *item.new.key)
+            for item in items
+            if not item.old.in_cache
+            and item.new.key != ROOT
+            and state != "added"
+        )
+    return output
+
+
+def diff(
+    out: "Output",
+    old_obj: Optional["HashFile"],
+    new_obj: Optional["HashFile"],
+    cache,
+    granular: bool = False,
+):
+    if granular:
+        return granular_diff(out, old_obj, new_obj, cache)
+    return shallow_diff(out, old_obj, new_obj, cache)
+
+
+class CmdDataStatus(CmdBase):
+    COLORS = {
+        "not_in_cache": "red",
+        "unstaged": "green",
+        "staged": "yellow",
+        "untracked": "cyan",
+    }
```

Comment:
```suggestion
    COLORS = {
        "not_in_cache": "red",
        "unstaged": "yellow",
        "staged": "green",
        "untracked": "cyan",
    }
```

Reply:
I don't think they all need unique colors. In Git, both untracked and unstaged are red. Can we do the same: make everything red except for staged/committed? WDYT?

Reply:
Missed your comment above. I'm fine with whatever you decide.